components/script/script_module.rs

```diff
@@ -657,6 +639,161 @@ impl ModuleTree {
         // Step 9.
         return errors.into_iter().max();
     }
+
+    #[allow(unsafe_code)]
+    /// https://html.spec.whatwg.org/multipage/#fetch-the-descendants-of-a-module-script
+    fn fetch_module_descendants(
+        &self,
+        owner: &ModuleOwner,
+        destination: Destination,
+        credentials_mode: CredentialsMode,
+        parent_identity: ModuleIdentity,
+    ) {
+        debug!("Start to load dependencies of {}", self.url.clone());
+
+        let global = owner.global();
+
+        self.set_status(ModuleStatus::FetchingDescendants);
+
+        match self.resolve_requested_modules(&global) {
+            // Step 3.
+            Ok(requested_urls) if requested_urls.len() == 0 => {
+                self.advance_finished_and_link(&global);
+            },
+            Ok(mut requested_urls) => {
+                self.descendant_urls
+                    .borrow_mut()
+                    .extend(requested_urls.clone());
+
+                let parent_identities = self.parent_identities.borrow();
+
+                for parent_identity in parent_identities.iter() {
+                    match parent_identity {
+                        ModuleIdentity::ScriptId(_) => continue,
+                        ModuleIdentity::ModuleUrl(parent_url) => {
+                            if requested_urls.contains(&parent_url.clone()) {
+                                self.advance_finished_and_link(&global);
```

Comment:
This might be quadratic if there are multiple matching URLs: we should set a boolean and run this algorithm only once outside the parent_identities loop